diff --git a/node_modules/react-native-ios-utilities/.DS_Store b/node_modules/react-native-ios-utilities/.DS_Store
new file mode 100644
index 0000000..84d2544
Binary files /dev/null and b/node_modules/react-native-ios-utilities/.DS_Store differ
diff --git a/node_modules/react-native-ios-utilities/react-native-ios-utilities.podspec b/node_modules/react-native-ios-utilities/react-native-ios-utilities.podspec
index 3d5ed71..a789fbb 100644
--- a/node_modules/react-native-ios-utilities/react-native-ios-utilities.podspec
+++ b/node_modules/react-native-ios-utilities/react-native-ios-utilities.podspec
@@ -54,6 +54,21 @@ Pod::Spec.new do |s|
     '"${PODS_ROOT}/Headers/Public/hermes-engine"',
     '"${PODS_ROOT}/Headers/Private/React-Core"',
     '"${PODS_CONFIGURATION_BUILD_DIR}/React-rendererconsistency/React_rendererconsistency.framework/Headers"',
+    # [begin] transitive dependencies of React-RCTAppDelegate that are not defined modules
+    '"${PODS_CONFIGURATION_BUILD_DIR}/React-RuntimeApple/React_RuntimeApple.framework/Headers"',
+    '"${PODS_CONFIGURATION_BUILD_DIR}/React-RuntimeCore/React_RuntimeCore.framework/Headers"',
+    '"${PODS_CONFIGURATION_BUILD_DIR}/React-jserrorhandler/React_jserrorhandler.framework/Headers"',
+    '"${PODS_CONFIGURATION_BUILD_DIR}/React-jsinspector/jsinspector_modern.framework/Headers"',
+    '"${PODS_CONFIGURATION_BUILD_DIR}/React-jsinspectortracing/jsinspector_moderntracing.framework/Headers"',
+    '"${PODS_CONFIGURATION_BUILD_DIR}/React-jsitooling/JSITooling.framework/Headers"',
+    '"${PODS_CONFIGURATION_BUILD_DIR}/React-runtimescheduler/React_runtimescheduler.framework/Headers"',
+    '"${PODS_CONFIGURATION_BUILD_DIR}/React-performancetimeline/React_performancetimeline.framework/Headers"',
+    '"${PODS_CONFIGURATION_BUILD_DIR}/React-rendererconsistency/React_rendererconsistency.framework/Headers"',
+    '"${PODS_CONFIGURATION_BUILD_DIR}/RCT-Folly/folly.framework/Headers"',
+    '"${PODS_CONFIGURATION_BUILD_DIR}/fmt/fmt.framework/Headers"',
+    '"${PODS_CONFIGURATION_BUILD_DIR}/React-utils/React_utils.framework/Headers"',
+    '"${PODS_CONFIGURATION_BUILD_DIR}/React-debug/React_debug.framework/Headers"',
+    '"${PODS_CONFIGURATION_BUILD_DIR}/React-rendererdebug/React_rendererdebug.framework/Headers"',
   ]
 
   # Swift/Objective-C compatibility
diff --git a/node_modules/react-native-ios-utilities/src/native_components/RNIDetachedView/RNIDetachedView.tsx b/node_modules/react-native-ios-utilities/src/native_components/RNIDetachedView/RNIDetachedView.tsx
index 5f7eb50..53c9050 100644
--- a/node_modules/react-native-ios-utilities/src/native_components/RNIDetachedView/RNIDetachedView.tsx
+++ b/node_modules/react-native-ios-utilities/src/native_components/RNIDetachedView/RNIDetachedView.tsx
@@ -68,6 +68,9 @@ export const RNIDetachedView = React.forwardRef<
   const reactChildrenCount = React.Children.count(props.children);
 
   const children = React.Children.map(props.children, (child) => {
+    if (React.isValidElement(child) && child.type === React.Fragment) {
+      return child;
+    }
     return React.cloneElement(
       child as React.ReactElement<RNIDetachedViewContentProps>, 
       {
